src/stm32/stm32f0_i2c.c

```diff
@@ -146,13 +146,24 @@ i2c_setup(uint32_t bus, uint32_t rate, uint8_t addr)
         enable_pclock((uint32_t)i2c);
         gpio_peripheral(ii->scl_pin, ii->function | GPIO_OPEN_DRAIN, 1);
         gpio_peripheral(ii->sda_pin, ii->function | GPIO_OPEN_DRAIN, 1);
-
-        // Set 100Khz frequency and enable
-        i2c->TIMINGR = ((0xB << I2C_TIMINGR_PRESC_Pos)
-                        | (0x13 << I2C_TIMINGR_SCLL_Pos)
-                        | (0xF << I2C_TIMINGR_SCLH_Pos)
-                        | (0x2 << I2C_TIMINGR_SDADEL_Pos)
-                        | (0x4 << I2C_TIMINGR_SCLDEL_Pos));
+        uint32_t pclk = get_pclock_frequency((uint32_t)i2c);
+        uint32_t presc = 1;
+        uint32_t scaled_pclk = pclk / (presc + 1);
+        // Minimal pclk 48Mhz stm32f0
+        while (scaled_pclk / rate > 255) {
```

Comment:
> Why this instead of just calculating it directly?

No real reason, just an artifact of the thought process here. 
Was not sure how it actually worked, so just came up with it.

AFAIK, `grep -R I2C_TIMINGR_` all headers show that we have 8 bit register for SCL{H,L}, and at least 4 bit register for {SDA,SCL}DEL.

I think we can decrease the threshold, but at a high level, it looks safe.
If that helps, we can calculate it from `0xff & (I2C_TIMINGR_SCLL >> I2C_TIMINGR_SCLL_pos) - 1`